homeassistant/components/switchbot/lock.py

```diff
@@ -0,0 +1,57 @@
+"""Support for SwitchBot lock platform."""
+from typing import Any
+
+import switchbot
+from switchbot import SwitchbotModel
+from switchbot.const import LockStatus
+
+from homeassistant.components.lock import LockEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import SwitchbotDataUpdateCoordinator
+from .entity import SwitchbotEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Switchbot lock based on a config entry."""
+    coordinator: SwitchbotDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    if coordinator.model == SwitchbotModel.LOCK:
+        async_add_entities([(SwitchBotLock(coordinator))])
```

Comment:
```suggestion
    async_add_entities([(SwitchBotLock(coordinator))])
```

This check should be unreachable since the lock platform won't ever be loaded if its not a lock